### ISSUE TITLE
Sweettea + Sugarcube 1.0.1 no longer works

### ISSUE THREAD

**Issue by goodwill** [colinta/teacup#106]
I got this with teacup 2.1.1 + Sugarcube 1.0.1 + Sweettea 0.5.6:

uninitialized constant Teacup::Layout (NameError)

**Comment by kjf** [comment]
Seeing the same issue with teacup 2.1.2 and Sugarcube 1.0.1.

**Comment by colinta** [comment]
Have you guys attempted anything to remedy this?  e.g. changing the order in your Gemfile (or using bundler in the first place)

Do you have `app.detect_dependencies` on or off?

**Comment by kjf** [comment]
@colinta I've not got it declared so whatever it defaults to. Should I have this enabled?

**Comment by colinta** [comment]
it's on by default.  i'm having trouble reproducing this error.

**Comment by goodwill** [comment]
Have been using bundler, no haven't tried app.detect_dependencies. Maybe you could come up with a very quick demo project so we can compare ourself?

**Comment by kjf** [comment]
@colinta Ok, so adding app.detect_dependencies = false seems get everything working again for me.

**Comment by colinta** [comment]
@goodwill Let me rephrase what you said above: you would like me to create and upload a small project that is compiling correctly for me so that you can download it and compile it and tell me it works or doesn't work so I can fix it.

That's asking too much, turn that around on its head: if you can create a small project that reproduces this error, I am much more likely to spend time on fixing it.

Teacup is not something that pays for my family's health insurance or savings, know what I'm saying?

`</rant>`

**Comment by kjf** [comment]
Hey folks, I made a simple app that should outline the issue. https://github.com/kjf/actionsheet-issue

If you bundle and then rake you should notice the app crash on launch with the error. **\* Terminating app due to uncaught exception 'NameError', reason: 'uninitialized constant Teacup::Layout (NameError)
'

**Comment by colinta** [comment]
Cool thanks, I do get the error, I'm attempting a fix by setting `app.files_dependencies` directly.  I wonder if Watson and Laurent have something to say (this might be a 2.4 or 2.5 regression)

see f66bf01, and update to it using

```
gem 'teacup', git: 'git@github.com:rubymotion/teacup'
```

**Comment by goodwill** [comment]
Relax, I thought you already have something on hand, thats why I would say its easier for us to just take a look on that.

And you know what? Miracle happened. Upgraded to sugarcube 1.0.3 and the problem is gone. I think something to do with how sugarcube detects if teacup has loaded? 

However, I would definitely hope you could help to look at issue #88

https://github.com/rubymotion/teacup/issues/88

I have created that project with the link attached, it definitely shows if I dont use teacup stylesheet controller got released immediately after I pop back to the previous controller. However if teacup is in use the controller is never released. Any idea? I tried to trace the source a bit but the stylesheet code is not easy for me to understand enough to find out where the reference is holding and preventing a proper release.

</love>

And I know it means nothing, but for your hardwork we can definitely donate to buy you a bear even if not paying your insurance. ;)

Re: ok I just double checked, its   app.detect_dependencies = false does the trick.

**Comment by colinta** [comment]
Right on, heheh, thanks for the kind words.  Managing a large project is a double edged sword.  You want people to use it, but once they do you start getting an outpouring of support requests!

**Comment by goodwill** [comment]
No worries, well understood. If you have a donate page I am sure quite a few people will buy you some beers :)
